src/main/java/com/ginsberg/gatherers4j/BigDecimalGatherer.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Todd Ginsberg
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ginsberg.gatherers4j;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.function.Function;
+import java.util.stream.Gatherer;
+
+import static com.ginsberg.gatherers4j.GathererUtils.mustNotBeNull;
+
+abstract public sealed class BigDecimalGatherer<INPUT>
+        implements Gatherer<INPUT, BigDecimalGatherer.State, BigDecimal>
+        permits BigDecimalSimpleAverageGatherer, BigDecimalSimpleMovingAverageGatherer, BigDecimalStandardDeviationGatherer {
```

Comment:
Why does this need to be sealed?

Reply:
It doesn't! I had it that way during a refactor and forgot to come back and take it out. Nice catch.